src/pip/_internal/req/req_install.py

```diff
@@ -811,6 +811,18 @@ def install(
         install_options = list(install_options) + self.install_options
 
         try:
+            if self.legacy_install_reason == 8559:
+                deprecated(
+                    reason=(
+                        f"{self.name} is being installed using the legacy "
+                        f"'setup.py install' method, because it does not have a "
+                        f"'pyproject.toml' and the 'wheel' package "
+                        f"is not installed."
+                    ),
+                    replacement="to enable the '--use-pep517' option",
```

Comment:
@pypa/pip-committers what do you think about the suggested replacement. Should we also suggest to install `wheel` - it is a transitory measure, which is why I did not add it for now, but may also be necessary if using `--no-build-isolation`.

Reply:
I'm inclined not to say install wheel. That's basically implied as an option by the deprecation reason itself. And as you say, it's not the approach we want to encourage.